operator/internal/manifests/config.go

```diff
@@ -169,7 +169,9 @@ func ConfigOptions(opt Options) config.Options {
 		},
 		StorageDirectory: dataDirectory,
 		MaxConcurrent: config.MaxConcurrent{
-			AvailableQuerierCPUCores: int32(opt.ResourceRequirements.Querier.Requests.Cpu().Value()),
+			// Here we multiply by 2 since with TSDB index dispatches many more,
+			// but each individually smaller, requests VS boltdb_shipper.
+			AvailableQuerierCPUCores: int32(opt.ResourceRequirements.Querier.Requests.Cpu().Value()) * 2,
```

Comment:
I believe we should remove this for now and evaluate proper values per index type in a separate PR.